lib/smart_proxy_container_gateway/container_gateway_main.rb

```diff
@@ -1,14 +1,22 @@
 require 'net/http'
 require 'uri'
 require 'digest'
-require 'sequel'
+require 'smart_proxy_container_gateway/dependency_injection'
 module Proxy
   module ContainerGateway
     extend ::Proxy::Util
     extend ::Proxy::Log
 
-    class << self
-      Sequel.extension :migration, :core_extensions
+    class ContainerGatewayMain
+      extend ::Proxy::ContainerGateway::DependencyInjection
+
+      #inject_attr :database_impl, :database
+
+      # TODO: make this method unnecessary by figuring out why container_instance is nil during dependency injection.
+      def database
```

Comment:
Not sure why the following is happening even though the dependency injection is set up the same as for the API class:

```
2024-04-02T21:58:01 642ac360 [W] Error processing request '642ac360-9cd3-49d9-9190-8ec249f638fe: <NoMethodError>: undefined method `get_dependency' for nil:NilClass
/usr/share/foreman-proxy/lib/proxy/dependency_injection.rb:74:in `block in inject_attr'
/usr/share/gems/gems/smart_proxy_container_gateway-1.2.0/lib/smart_proxy_container_gateway/container_gateway_main.rb:191:in `insert_token'
/usr/share/gems/gems/smart_proxy_container_gateway-1.2.0/lib/smart_proxy_container_gateway/container_gateway_api.rb:144:in `block in <class:Api>'
/usr/share/gems/gems/sinatra-2.2.4/lib/sinatra/base.rb:1697:in `call'
/usr/share/gems/gems/sinatra-2.2.4/lib/sinatra/base.rb:1697:in `block in compile!'

```

Reply:
I've created this `database` method to work around it.

Reply:
If the reviewer has any idea what is the difference in calling inject_attr in container_gateway_main vs container_gateway_api, I'd love to hear. At this point my only thought is that it has something to do with how the main class is a dependency of the api class.

Reply:
If you look at https://github.com/ekohl/smart_proxy_container_gateway/commit/ae839c1511a73724ec0df6ebe242310323ea2fb9 then I implemented it without dependency injection in this class.

Specifically, in https://github.com/ekohl/smart_proxy_container_gateway/commit/ae839c1511a73724ec0df6ebe242310323ea2fb9#diff-b13ec686c62aea565146e734f41d2d2e3ee8c0a6258fccf3aefaad7ef4ba80c5R10-R16 I have a constructor that sets parameters.

Then when wiring up the dependencies this is created: https://github.com/ekohl/smart_proxy_container_gateway/commit/ae839c1511a73724ec0df6ebe242310323ea2fb9#diff-01554a5e39079ab23f2a4fbb74dfb55fce9e373d490b40aae27811c6237df2e3R37-R42

I'd expect that to work, but I haven't tested it myself.

Reply:
Ahh okay, yeah that might be the missing last refactoring I need.